introduction_to_amazon_algorithms/jumpstart-foundation-models/retrieval_augmented_generation_question_answering_embedding.ipynb

```diff
@@ -0,0 +1,688 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 4.1 Deploying the model endpoint for GPT-J-6B embedding model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model_id, model_version = \"huggingface-textembedding-gpt-j-6b\", \"*\"\n",
+    "\n",
+    "endpoint_name_embed = name_from_base(f\"jumpstart-example-{model_id}\")\n",
+    "\n",
+    "\n",
+    "embed_model = Model(\n",
+    "    image_uri=\"763104351884.dkr.ecr.us-west-2.amazonaws.com/djl-inference:0.21.0-deepspeed0.8.0-cu117-v1.0\",\n",
+    "    model_data=\"s3://gptjfinetune/script_embed/xinembed.tar.gz\",\n",
+    "    role=aws_role,\n",
+    "    predictor_cls=Predictor,\n",
+    "    name=endpoint_name_embed,\n",
+    ")\n",
+    "\n",
+    "model_predictor_embed = embed_model.deploy(\n",
+    "    initial_instance_count=1,\n",
+    "    instance_type=inference_instance_type,\n",
+    "    predictor_cls=Predictor,\n",
+    "    endpoint_name=endpoint_name_embed,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "endpoint_name_embed = \"jumpstart-example-huggingface-textembed-2023-03-24-21-04-00-882\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from tqdm import tqdm\n",
+    "\n",
+    "\n",
+    "def parse_response_text_embed(query_response):\n",
+    "    model_predictions = json.loads(query_response[\"Body\"].read())\n",
+    "    return model_predictions[\"generated_embed\"][0]\n",
+    "\n",
+    "\n",
+    "def build_embed_table(df_knowledge, endpoint_name_embed, col_name_4_embed):\n",
+    "    res_embed = []\n",
+    "    N = df_knowledge.shape[0]\n",
+    "    for idx in tqdm(range(N)):\n",
+    "        content = df_knowledge.loc[idx][col_name_4_embed]\n",
+    "        query_response = query_endpoint_with_json_payload(\n",
+    "            content, endpoint_name_embed, content_type=\"application/x-text\"\n",
+    "        )\n",
+    "        generated_embed = parse_response_text_embed(query_response)\n",
+    "        res_embed.append(generated_embed)\n",
+    "    res_embed_df = pd.DataFrame(res_embed)\n",
+    "    return res_embed_df"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 4.2. Generate embedings for each of document in the knowledge library with the GPT-J-6B embedding model.\n",
+    "\n",
+    "We use the answers of FAQ sections in Amaon SageMaker documents as the knowledge library. The dataset is defined as below. The `Question` column will be used as reference for you to ask a question. The `Answer` column contains documents of knowledge library. We will iterate each document to get its embedding vector via the GPT-J-6B embedding models."
```

Comment:
I don't understand the following sentence: "The `Question` column will be used as reference for you to ask a question." consider rephrasing.

Reply:
resolved in next revision